website/docs/guides/migration/versions/04-upgrading-to-dbt-utils-v1.0.md

```diff
@@ -0,0 +1,140 @@
+- All other cross-db macros have moved to the dbt namespace, with no changes necessary other than replacing `dbt_utils.` with `dbt.`. Review the [cross database macros documentation](https://docs.getdbt.com/reference/dbt-jinja-functions/cross-database-macros) for the full list.
+    - In your code editor, you can do a global find and replace with regex: `\{\{\s*dbt_utils\.(any_value|bool_or|cast_bool_to_text|concat|dateadd|datediff|date_trunc|escape_single_quotes|except|hash|intersect|last_day|length|listagg|position|replace|right|safe_cast|split_part|string_literal|type_bigint|type_float|type_int|type_numeric|type_string|type_timestamp|type_bigint|type_float|type_int|type_numeric|type_string|type_timestamp|except|intersect|concat|hash|length|position|replace|right|split_part|escape_single_quotes|string_literal|any_value|bool_or|listagg|cast_bool_to_text|safe_cast|dateadd|datediff|date_trunc|last_day)` → `{{ dbt.$1`
+- The `insert_by_period` materialization has been moved to the [experimental-features repo](https://github.com/dbt-labs/dbt-labs-experimental-features/tree/main/insert_by_period). To continue to use it, add the below to your packages.yml file:
+
+```yaml
+packages:
+  - git: https://github.com/dbt-labs/dbt-labs-experimental-features
+    subdirectory: insert_by_period
+    revision: XXXX #optional but highly recommended. Provide a full git sha hash, e.g. 1c0bfacc49551b2e67d8579cf8ed459d68546e00. If not provided, uses the current HEAD.
+```
+
+- `safe_add()` only works with a list of arguments; use `{{ dbt_utils.safe_add(['column_1', 'column_2']) }}` instead of varargs `{{ dbt_utils.safe_add('column_1', 'column_2') }}`.
+- Several long-promised deprecations to `deduplicate()` have been applied:
+    - The `group_by` argument is replaced by `partition_by`.
+    - `relation_alias` is removed. If you need an alias, you can pass it directly to the `relation` argument.
+    - `order_by` is now mandatory. Pass a static value like `1` if you don’t care how they are deduplicated.
+- The deprecated `table` argument has been removed from `unpivot()`. Use `relation` instead.
+
+
+## Resolving error messages
+After upgrading, these are common error messages you may encounter, along with their resolutions.
+<details>
+	<summary><code>dict object has no attribute MACRO_NAME</code></summary>
+	<div>
+		<p><b>Cause</b>: No macro called <code>MACRO_NAME</code> exists. This is most likely because the macro has moved to the <code>dbt</code> namespace (see above). It could also be because you haven't run dbt deps or have misspelled a macro's name.</p>
+		<p><b>Resolution</b>: For <a href="/reference/dbt-jinja-functions/cross-database-macros">cross-database macros</a>, change <code>dbt_utils.MACRO_NAME()</code> to <code>dbt.MACRO_NAME()</code>.</p>
+	</div>
+</details>
+<details>
+	<summary><code>macro 'dbt_macro__generate_surrogate_key' takes not more than 1 argument(s)</code> </summary>
```

Comment:
```suggestion
	<summary>The macro <code>dbt_macro__generate_surrogate_key</code> takes not more than 1 argument(s) </summary>
```

Reply:
This is the exact error message and format, so can't change it